src/or/dirauth/dirvote.c

```diff
@@ -248,8 +249,12 @@ format_networkstatus_vote(crypto_pk_t *private_signing_key,
       params = smartlist_join_strings(v3_ns->net_params, " ", 0, NULL);
     else
       params = tor_strdup("");
-
     tor_assert(cert);
+    if (v3_ns->bwlist_headers)
+      bwlist_headers = smartlist_join_strings(v3_ns->bwlist_headers, " ", 0,
+                                              NULL);
```

Comment:
Please add a limit to the length of the bwlist_headers string. I don't expect we will ever have more than 1024 (~50*20) bytes of headers. Truncate the string when the limit is reached.